File: gachisa-backend/src/main/java/com/gachisa/payment/dto/GroupBuyResultCommand.java
package com.gachisa.payment.dto;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.util.List;

public record GroupBuyResultCommand(
        @NotNull Long groupBuyId,
        @NotNull Result result,
        @NotEmpty @Size(max = 1000) List<@NotNull Long> participationIds
) {

    public enum Result {
        ACHIEVED,
        FAILED
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/payment/dto/GroupBuyResultCommand.kt
package com.gachisa.payment.dto

import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class GroupBuyResultCommand(
    @field:NotNull val groupBuyId: Long,
    @field:NotNull val result: Result,
    @field:NotEmpty @field:Size(max = 1000) val participationIds: List<@NotNull Long>,
) {
    enum class Result {
        ACHIEVED,
        FAILED,
    }
}
